### PR TITLE
Update to pip==19

### LINKED ISSUE #2910: Docker image build fails
Building the `olweb` image no longer works and which blocks all development because it needs to be rebuilt to include the new `deprecated` dependency.

### Evidence / Screenshot (if possible)

```
Collecting zipp>=0.5 (from importlib-metadata>=0.12; python_version < "3.8"->pytest==4.6.6->-r requirements_test.txt (line 9))
  Downloading https://files.pythonhosted.org/packages/ee/f7/6c3ca4764a8025da9422825c2bc367fdc8dd115ebb52e4ef057f234938c4/zipp-2.0.0.tar.gz
  Running setup.py (path:/tmp/pip-build-j64QQ0/zipp/setup.py) egg_info for package zipp produced metadata for project name unknown. Fix your #egg=zipp fragments.

```
This is related to: https://github.com/jaraco/zipp/issues/14

### Proposal & Constraints

I've confirmed that updating to pip=9 will fix the problem. (The version that we're using from xenial is 8 and the current version is 20).

I'll generate a PR with the fix.

### Stakeholders
@hornc @cdrini @mekarpeles

### PR DESCRIPTION
**Pull request by tfmorris**
Fixes #2910

Allows us to build Docker images again as a stopgap until with update to Bionic.

@hornc @mekarpeles

### CODE REVIEW

File: docker/Dockerfile.olbase [hunk 1]
```diff
@@ -29,6 +29,9 @@ RUN apt-get -qq update && apt-get install -y \
     python-psycopg2 \
     python-yaml
 
+# Update pip to at least 9.0
+RUN pip install --upgrade pip==9
```
Comment (cdrini) [inline docker/Dockerfile.olbase]: The version on ol-dev0 and ol-web3 is 19.3.1

```suggestion
RUN pip install --upgrade pip==19.3.1
```

Comment (cclauss) [reply docker/Dockerfile.olbase]: Why not `pip==20.0.1`?

Comment (cdrini) [reply docker/Dockerfile.olbase]: That's what's on our production environment. We don't have a great way of updating things there yet (they're not running docker, so we have to update each dep manually, which is a time sink for a 2.5 person team). I think keeping things in sync is more beneficial than keeping things bleeding edge; once we have docker in more places, updating dependencies will be much easier.

**Comment by tfmorris** [comment]
Seems like someone who had access to the secret production environment would be in a better position that me to make sure these track, but I've made the requested change.

pip 19 appears to be stricter about uninstalling packages that it didn't install, so I removed the apt-get packages that get overwritten later by pip.

I also fixed a couple of other build errors related to file permissions and missing home directories (e.g. `npm update` didn't work).

Adding `node_modules` to `.dockerignore` is a performance optimization (since I had to iterate so many times on the build).

**Comment by tfmorris** [comment]
I'm not super motivated to investigate in detail, but when I checked a week ago, `zipp` had two major version bumps in January, which I consider a smoking gun. It's likely that the new release is incompatible with the ancient version of pip that we're currently using. It's also possible that some other dependency update triggered this transient dependency issue.

If you read the conversation in the issue that I linked to the original problem report, you can see that the incompatibility has been known for a while and the zipp team has no plans to address it when a simple pip upgrade fixes the issue.

File: docker/Dockerfile.olbase [hunk 2]
```diff
@@ -43,20 +43,20 @@ RUN ln -s /openlibrary/conf/nginx/sites-available/openlibrary.conf /etc/nginx/si
...
+USER openlibrary
 COPY package*.json ./
 RUN npm install
 
 COPY --chown=openlibrary:openlibrary . /openlibrary
 
-USER openlibrary
 RUN ln -s vendor/infogami/infogami infogami
 # run make to initialize git submodules, build css and js files
+RUN npm update caniuse-lite browserslist
```
[the remark this answers is not in the document]
Comment (cdrini) [reply docker/Dockerfile.olbase]: Ah, looks like something even odder is going on; the deps _do_ look outdated, but calling update doesn't edit the `package-lock.json`? Going to attribute this to our old version of node/npm, so should be addressed in the other PR, but removed from here.

**Comment by cdrini** [comment]
Darn! Messed that up; one sec and I'll fix it.

File: docker/Dockerfile.olbase (hunk 2 above)
[the remark this answers is not in the document]
Comment (cdrini) [reply docker/Dockerfile.olbase]: Done

File: docker/Dockerfile.olbase [hunk 3]
```diff
@@ -7,7 +7,7 @@ ENV LC_ALL POSIX
 
 # add openlibrary users
 RUN groupadd --system openlibrary \
-  && useradd --no-log-init --system --gid openlibrary openlibrary \
+  && useradd --no-log-init --system --gid openlibrary --create-home --home /openlibrary openlibrary \
```
[the remark this answers is not in the document]
Comment (cdrini) [reply docker/Dockerfile.olbase]: Done

**Comment by cdrini** [comment]
It was easier to just make the requisite changes than to pull out the pip update.

Tested:
- ✅ Doing a full rebuild (after even deleting the xenial image) doesn't error
  - It did show the caniuselist warning, but as noted I think that's a node/npm version issue.
- ✅ Resulting image loads the site
- ✅ `docker-compose exec web test` works

**Comment by tfmorris** [comment]
@cdrini Is that working the way you want? You changed the home directory, but none of the other commands that rely on it, so I'm suspicious. For example, won't the node installation now go to `/home/openlibrary` instead of `/openlibrary` as a result of the USER command?

**Pull request merged**